packages/connection/src/Manager.php

```diff
@@ -1735,4 +1735,24 @@ public function sign_role( $role, $user_id = null ) {
 
 		return $role . ':' . hash_hmac( 'md5', "{$role}|{$user_id}", $token->secret );
 	}
+
+	/**
+	 * Some hosts disable the OpenSSL extension and so cannot make outgoing HTTPS requsets
+	 */
+	public static function fix_url_for_bad_hosts( $url ) {
+		if ( 0 !== strpos( $url, 'https://' ) ) {
```

Comment:
let's rewrite this method, I think a simple `Defined JETPACK_CLIENT__HTTPS and JETPACK_CLIENT__HTTPS === JETPACK_CLIENT__HTTPS then set_url_scheme( $url, 'http' )` would achieve the same?

Reply:
Did you mean to suggest comparing the constant to itself `JETPACK_CLIENT__HTTPS === JETPACK_CLIENT__HTTPS`, or did you have something else in mind? If you did, I don't follow.

This function doesn't make a ton of sense to me. 
```
public static function fix_url_for_bad_hosts( $url ) {
		if ( 0 !== strpos( $url, 'https://' ) ) {
			return $url;
		}

		switch ( JETPACK_CLIENT__HTTPS ) {
			case 'ALWAYS':
				return $url;
			case 'NEVER':
				return set_url_scheme( $url, 'http' );
		}

		// we now return the unmodified SSL URL by default, as a security precaution
		return $url;
	}
```

It seems to be saying:
1) If we receive a URL without https, return that url.
2) If we receive a URL and `JETPACK_CLIENT__HTTPS` is set to `ALWAYS`, return the url whether or not it's https. I suppose it will necessarily be https at this point since step 1 returned anything non https. At that rate, what is the point of having this case?
3) If we receive a URL and `JETPACK_CLIENT__HTTPS` is `NEVER`, it converts an https url to non https and returns it.
4) Otherwise, it returns the original url. I think this should be the default case in the switch statement instead of a final return statement.

In light of the above, couldn't we rewrite the method as 

```
public static function fix_url_for_bad_hosts( $url ) {
    // If we receive an http url, return it
    if ( 'http' === wp_parse_url( $url, PHP_URL_SCHEME ) ) {
        return $url;
     } 

    // If the url should never be https, ensure it isn't https
    if ( 'NEVER' === JETPACK_CLIENT__HTTPS ) {
        return set_url_scheme( $url, 'http' );
    }

    // Otherwise, return the https url
    return $url
}
```

What do you think?

Reply:
Sounds good, and looks much clearer. Let's use the `Constants` package in the rewrite.

Reply:
See a142b9c. The `fix_url_for_bad_hosts` method was rewritten as suggested by @gititon, and the Constants package was used.